payments-core/src/main/java/com/stripe/android/paymentsheet/injection/PaymentSheetViewModelModule.kt

```diff
@@ -0,0 +1,224 @@
+            stripeAccountId = lazyPaymentConfig.get().stripeAccountId,
+            workContext = Dispatchers.IO
+        )
+    }
+
+    @Provides
+    @Singleton
+    fun providePaymentFlowResultProcessor(
+        application: Application,
+        starterArgs: PaymentSheetContract.Args,
+        lazyPaymentConfig: Lazy<PaymentConfiguration>,
+        stripeApiRepository: StripeApiRepository,
+    ): PaymentFlowResultProcessor<out StripeIntent, StripeIntentResult<StripeIntent>> {
+        return when (starterArgs.clientSecret) {
+            is PaymentIntentClientSecret -> PaymentIntentFlowResultProcessor(
+                application,
+                { lazyPaymentConfig.get().publishableKey },
+                stripeApiRepository,
+                enableLogging = true,
+                Dispatchers.IO
+            )
+            is SetupIntentClientSecret -> SetupIntentFlowResultProcessor(
+                application,
+                { lazyPaymentConfig.get().publishableKey },
+                stripeApiRepository,
+                enableLogging = true,
+                Dispatchers.IO
+            )
+        }
+    }
+
+    @Provides
+    @Singleton
+    fun provideGooglePayRepository(
+        application: Application,
+        starterArgs: PaymentSheetContract.Args
+    ): GooglePayRepository {
+        return starterArgs.config?.googlePay?.environment?.let { environment ->
+            DefaultGooglePayRepository(
+                application,
+                environment
+            )
+        } ?: GooglePayRepository.Disabled
+    }
+
+    @Provides
+    @Singleton
+    fun providePrefsRepository(
+        application: Application,
+        starterArgs: PaymentSheetContract.Args,
+        googlePayRepository: GooglePayRepository
+    ): PrefsRepository {
+        return starterArgs.config?.customer?.let { (id) ->
+            DefaultPrefsRepository(
+                application,
+                customerId = id,
+                isGooglePayReady = { googlePayRepository.isReady().first() },
+                workContext = Dispatchers.IO
+            )
+        } ?: PrefsRepository.Noop()
+    }
+
+    // TODO: Replace with an actual logger
+    @Provides
+    @Singleton
+    fun provideLogger() = Logger.noop()
+
+    @Provides
+    @Singleton
+    @Named(WORK_CONTEXT)
+    fun provideWorkContext(): CoroutineContext = Dispatchers.IO
+
+    /**
+     * Provides the [StripePaymentController] instance.
+     *
+     * Use [Lazy] to delay [PaymentSheetViewModel] instance fetch to resolve its circular
+     * dependency with [StripePaymentController]
+     *
+     * TODO(ccen): abstract onPaymentFlowResultCallback to avoid passing [PaymentSheetViewModel]
+     */
+    @Provides
+    @Singleton
+    fun provideStripePaymentController(
+        application: Application,
+        stripeApiRepository: StripeApiRepository,
+        activityLauncherFactory: ActivityLauncherFactory,
```

Comment:
I think a comment was brought up in chat to have this be just the activity rather than a factory

Reply:
yes, resolved by #3834